tensorflow_serving/batching/batching_util.cc

```diff
@@ -0,0 +1,229 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_serving/batching/batching_util.h"
+
+#include <string>
+
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/types.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+
+namespace tensorflow {
+namespace serving {
+
+// Padding for one dimension of a tensor.
+struct OneDimPadding {
+  int64 first;  // pad before
+  int64 second;  // pad after
+};
+
+// Constructs array of paddings, where:
+// paddings[i].first - number of objects to add before elements in dimension i
+// of given tensor,
+// paddings[i].second - number of objects to add after elements in dimension i.
+// This padding signature is used when performing internal padding with Eigen.
+//
+// When building paddings it is assumed that tensor needs to be padded
+// after each dimension, so its shape matches max_dim_sizes,
+// First entry of max_dim_sizes, which is maximum size of zeroth dimension,
+// is ignored, because we don't perform padding in that dimension.
+template<int num_dims>
+std::array<OneDimPadding, num_dims> CreatePadding(Tensor tensor,
```

Comment:
Well, I pass the value returned by this function to pad method of Eigen::TensorMap. When I try to pass std::vector instead of std::array it gives me compilation error saying "template instantiation depth exceeds maximum of 900" on this line. There appears to be some recursive template array unfolding. It seems that Eigen needs to know size of array at compile time.